docs/xbcloud-binary-fifo-datasink.md

```diff
@@ -0,0 +1,58 @@
+# FIFO datasink
+
+[Percona XtraBackup 8.0.33-28](release-notes/8.0/8.0.33-28.0.md) implements a new datasink that uses `FIFO` (named pipes). With FIFO datasink feature, you can benefit from faster backups by streaming data in parallel from XtraBackup to `xbcloud`.
+
+## Stream to object storage
+
+When taking a backup, you can save the files locally or stream the files to either different servers or object storage. 
+
+Streaming backups to object storage using [`xbcloud`](xbcloud-binary-overview.md) works as follows:
+
+* XtraBackup spawns multiple copy threads. 
+
+* Each copy thread reads a chunk of data from a specific file.
+
+* Each copy thread writes the chunk of data to a pipe (STDOUT).
+
+* A read thread reads each chunk of STDIN (standard input device) data. xbcloud uploads the chunks to an object storage using an async request, and a callback is added to an event handler list.
+
+* The xbcloud event handler checks when the TCP socket receives the response from the object storage and executing the callback depending on the return (success or failure).
+
+![image](_static/backup-streamed-to-object-storage.png)
+
+The speed of XtraBackup streaming has a limitation of around 1.8 Gbps. This happens because XtraBackup writes only one thread to STDOUT simultaneously via a mutex to ensure that a chunk of data is written from the beginning till the end. This architecture works fine when uploading data via WAN. However, for streaming backups using LAN, the most popular setup is 10 Gbps or more.
```

Comment:
XtraBackup streaming has a limitation of 1.8 Gbps. XtraBackup writes one thread to STDOUT simultaneously using a mutex. This operation ensures that a chunk is written from the beginning to the end. 
Uploading chunks on a Wide Area Network (WAN) experiences no lag, but streaming backups using a Local Area Network (LAN) can experience a lag because those configurations are designed for 10Gbps.

Reply:
updated